### ISSUE TITLE
[FEATURE REQUEST] Additional Sort Order setting

### ISSUE THREAD

**Issue by mmattel** [owncloud/ios-app#975]
**Please describe.**
Files shown on IOS devices sorted by name look like this:

1_datei.txt
10_datei.txt
11_datei.txt
2_datei.txt
3_datei.txt

**Describe the solution you'd like**
The sort order should be additional selectable like this:

1_datei.txt
2_datei.txt
3_datei.txt
10_datei.txt
11_datei.txt

### TASKS

- [ ] Research (if needed)
- [ ] Create branch feature/feature_name
- [ ] Development tasks
  - [ ] Implement whatever
  - [ ] ...
- [ ] Code review and apply changes requested
- [ ] Design test plan
- [ ] QA
- [ ] Merge branch feature/feature_name into master

### PR

- App
- Library (if needed)

**Comment by michaelstingl** [comment]
Hm, good point. This is like in macOS Big Sur Finder and iOS 14 Files app.

![macOS Big Sur Finder](https://user-images.githubusercontent.com/214010/119699721-bd73a300-be52-11eb-893a-9169a62b2e31.png) | ![iOS 14 Files app](https://user-images.githubusercontent.com/214010/119699741-c2d0ed80-be52-11eb-882a-9418e0394b88.jpeg)
-- | --
macOS Big Sur Finder | iOS 14 Files app

@hosy @felix-schwarz do you remember why we choose `1_datei.txt`, `10_datei.txt`, `2_datei.txt` ?

**Comment by michaelstingl** [comment]
Same in oC 10.7 web UI and ownCloud-web:

<img width="573" alt="CleanShot 2021-05-26 at 18 57 07@2x" src="https://user-images.githubusercontent.com/214010/119701055-3b847980-be54-11eb-91c9-41563cdaa00e.png"> | <img width="648" alt="CleanShot 2021-05-26 at 18 57 48@2x" src="https://user-images.githubusercontent.com/214010/119701101-48a16880-be54-11eb-86b4-25448d9ed178.png">
-- | --
oC 10.7 web UI | ownCloud-web

I think we should change it.

**Comment by michaelstingl** [comment]
@mneuwert is this related to “collation”? What spec did we implement? Is there another rule that would be better?

**Comment by mneuwert** [comment]
@michaelstingl Sorting is implemented in the _ownCloudAppShared_ framework. In case of alphabetic sort, we just use Swift standard library API `String.caseInsensitiveCompare(String)`. Foundation `NSString.compare(options:)` would allow to tweak comparison algorithm behaviour by applying several available options like:

- ignore diacrtics 
- numeric comparison
- etc

There is even quite old dev Q/A from Apple discussing the topic of "comparing files name like in Finder" using CoreFoundation functions:
https://developer.apple.com/library/archive/qa/qa1159/_index.html

To summarise: this would be an easy fix. Just replacing `String.caseInsensitiveCompare()` with `NSString.compare() `with few options (numeric, case-insensitive, width-insensitive) would do the job IMHO

**Comment by jesmrec** [comment]
Done and released in 11.9
